protostuff-collectionschema/src/main/java/io/protostuff/MapSchema.java

```diff
@@ -47,6 +47,44 @@
         public Class<?> typeClass();
     }
 
+    private static final Comparable EMPTY_OBJECT = new Comparable()
+    {
+        @Override
+        public int compareTo(Object o)
+        {
+            return o == this ? 0 : -1;
+        }
+        @Override
+        public boolean equals(Object o)
+        {
+            return o == this;
+        }
+        @Override
+        public int hashCode()
+        {
+            return 0;
+        }
+    };
+
+    @SuppressWarnings("unchecked")
+    public static <T extends Map> T removeUnmodifiableWrapper(T map, MessageFactory messageFactory)
+    {
+        if(map == null)
+            return null;
+
+        try
+        {
+           map.put(EMPTY_OBJECT, EMPTY_OBJECT);
+           map.remove(EMPTY_OBJECT);
+        }

```

Comment:
Can you try setting this property and see if you still get the failures?
```
-Dprotostuff.runtime.always_use_sun_reflection_factory=true

// or via System.setProperty on your main class
```

A better solution is to mark the field with:
```
@Tag(value = 1, merge = false) // to be added.
Collection<String> foo = ...; // any immutable init you can think of
```
That way, the deserializer would always overwrite the field with a new one.